xdsl/utils/hints.py

```diff
@@ -42,10 +43,11 @@ def isa(arg: Any, hint: type[_T]) -> TypeGuard[_T]:
     if origin in [Union, UnionType]:
         return any(isa(arg, union_arg) for union_arg in get_args(hint))
 
-    from xdsl.irdl import GenericData
-    if (origin is not None) and issubclass(origin, GenericData):
+    from xdsl.irdl import GenericData, irdl_to_attr_constraint
+    if (origin is not None) and issubclass(
+            origin, GenericData | ParametrizedAttribute):
+        constraint = irdl_to_attr_constraint(hint)
         try:
```

Comment:
I feel like a need to point out once again that I do not like triggering exceptions as part of normal control flow. Maybe varify could instead return a bool? 😅

Reply:
I agree here, though this will need a redesign of the entire verification system (in another PR).
One way of doing this would be to return a `Literal[True] | Error`, or something similar.

Reply:
Sure, I am fine with postponing it 👍